homeassistant/components/sia/binary_sensor.py

```diff
@@ -24,44 +25,64 @@
     CONF_ZONES,
     SIA_HUB_ZONE,
     SIA_UNIQUE_ID_FORMAT_BINARY,
+    SIA_UNIQUE_ID_FORMAT_HUB,
 )
 from .sia_entity_base import SIABaseEntity
 
 _LOGGER = logging.getLogger(__name__)
 
-
-POWER_CODE_CONSEQUENCES: dict[str, bool] = {
-    "AT": False,
-    "AR": True,
+CODE_CONSEQUENCES: dict[str, dict[str, bool]] = {
+    DEVICE_CLASS_SMOKE: {
+        "GA": True,
+        "GH": False,
+        "FA": True,
+        "FH": False,
+        "KA": True,
+        "KH": False,
+    },
+    DEVICE_CLASS_MOISTURE: {
```

Comment:
Please consider implementing `EntityDescription`s instead of this logic.

Reply:
Done!